File: agent-sql-bun/src/cli/connection/list.ts
import type { Command } from "commander";
import { getConnections, getDefaultConnectionAlias } from "../../lib/config.ts";
import { printJson } from "../../lib/output.ts";

export function registerList(connection: Command): void {
  connection
    .command("list")
    .description("List saved connections")
    .action(() => {
      const connections = getConnections();
      const defaultAlias = getDefaultConnectionAlias();

      const items = Object.entries(connections).map(([alias, conn]) => ({
        alias,
        driver: conn.driver,
        host: conn.host,
        port: conn.port,
        database: conn.database,
        path: conn.path,
        url: conn.url,
        credential: conn.credential,
        default: alias === defaultAlias,
      }));

      printJson({ connections: items }, { prune: true });
    });
}

File: internal/cli/connection/list.go
package connection

import (
	"github.com/spf13/cobra"

	"github.com/shhac/agent-sql/internal/config"
	"github.com/shhac/agent-sql/internal/output"
)

func registerList(parent *cobra.Command) {
	list := &cobra.Command{
		Use:   "list",
		Short: "List saved connections",
		Args:  cobra.NoArgs,
		RunE: func(cmd *cobra.Command, args []string) error {
			conns := config.GetConnections()
			defaultAlias := config.GetDefaultAlias()

			items := make([]any, 0, len(conns))
			for alias, conn := range conns {
				items = append(items, conn.AsReceipt(alias, alias == defaultAlias))
			}

			output.PrintList(items, nil, true)
			return nil
		},
	}
	parent.AddCommand(list)
}
